### PR TITLE
Rough fixes to get to a working oauth 2 flow

### PR DESCRIPTION
**Pull request by jmullan**
I "needed" to push another playlist from my home server to rdio, so I did the minimum to make this work (I think)

### CODE REVIEW

**Comment by dasevilla** [comment]
Hi there! I haven't forgot about this, sorry for the delay

**Comment by jmullan** [comment]
There's a phabricator review for this, too.

File: .travis.yml [hunk 1]
```diff
@@ -1,6 +1,23 @@
+# Travis CI configuration file
```
Comment (dasevilla) [inline .travis.yml]: While your updating, can you do this? http://docs.travis-ci.com/user/migrating-from-legacy/#tl%3Bdr

File: rdio-call [hunk 2]
```diff
@@ -21,99 +21,108 @@ AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
...
+    if options.verbose:
+      if 'www-authenticate' in response:
+        print 'Authentication required, pass --authenticate'
+      else:
+        print content
+    sys.exit(int(response['status']))
+
+if __name__ == "__main__":
+  parser = OptionParser(usage='%prog [options] method arg1=value1 arg2=value2...', version='%prog 0.1')
+  parser.add_option('--client-key', dest='client_key',
```
Comment (dasevilla) [inline rdio-call]: Please call it _client ID_ instead of _client key_ to be consistent with the API docs

File: .travis.yml (hunk 1 above)
Comment (jmullan) [inline .travis.yml]: Done! We'll see if travis is happy again.

File: rdioapi/__init__.py [hunk 3]
```diff
@@ -20,139 +20,288 @@
...
+      'device_code_url': self.DEVICE_CODE_URL,
+      'token_url': self.TOKEN_URL
+    }
+    for key in self.urls:
+      if key in urls and urls[key] is not None:
+        self.urls[key] = urls[key]
+    self.client_id = client_id
+    self.client_secret = client_secret
+    self._store = AuthStore(data_store)
+    self._client = oauth2.Client(oauth2.Consumer(client_id, client_secret), None)
```
Comment (dasevilla) [inline rdioapi/__init__.py]: `oauth2` is confusingly an OAuth 1.0 library. We should be able to remove this

**Comment by jmullan** [comment]
Ugh, my pytravis script (to run the travis stuff locally) was broken, so I was not getting the lint errors.

**Comment by dasevilla** [comment]
Can you squash the commits?

File: rdio-call (hunk 2 above)
Comment (jmullan) [inline rdio-call]: Oh, I didn't get this note. I'll fix that (it was consumer_key before)

**Comment by jmullan** [comment]
Ok, commits squashed and consumer id turned to client id

**Comment by dasevilla** [comment]
Thanks! :cake:

**Pull request merged**